### DISCUSSION TITLE
Select Form Component Toggle Disable Enable

### DISCUSSION THREAD

**Discussion by untillnesss** [filamentphp/filament#9901] [Help] [answered]
![image](https://github.com/filamentphp/filament/assets/33585758/5508c53f-0569-45c9-b84d-48d5e31014fd)

I have a Select Form Component, and I gave it Disable and Enable properties based on other fields, Disable and Enable succeeded in changing according to the contents of the field, but the Select Component Field cannot be clicked or opened.

![image](https://github.com/filamentphp/filament/assets/33585758/03ceafb5-7f0b-4f67-acfb-5efe27a532f0)

If I remove the disabled function, it can run fine.

![image](https://github.com/filamentphp/filament/assets/33585758/c49ffb90-b717-4809-b59b-5416042e5232)

![image](https://github.com/filamentphp/filament/assets/33585758/178f7a99-f297-4a32-9657-75a9b14d5fcd)

I wonder why ?

**Comment by leandrocfe** [comment]
You should add `->live()` to the `class_stage_id` field

```php
Select::make('class_stage_id')
    ->live()
    ->options([
        1 => 'Option 1',
    ]),
Select::make('class_room_id')
    ->options([
        2 => 'Option 2',
        3 => 'Option 3',
        4 => 'Option 4',
    ])
    ->disabled(fn (Get $get): ?bool => ! $get('class_stage_id'))
```

**Reply by untillnesss** [reply]
class_stage_id already has a ->live() parameter and Toggle Enable Disable works fine, but Select Component becomes unclickable. Is there an error in the filament?

**Reply by leandrocfe** [reply]
Share the whole code please

**Reply by untillnesss** [reply]
Here it is.

```php
Select::make('class_stage_id')
    ->options(ClassStage::where('is_active', 1)->get()->pluck('value', 'id'))
    ->optionsLimit(12)
    ->preload()
    ->searchable()
    ->required()
    ->live()
    ->columnSpanFull()
    ->afterStateUpdated(function ($set) {
        $set('class_room_id', null);
    }),

Select::make('class_room_id')
    ->options(function ($get) {
        return
            ClassRoom::query()
            ->where('is_active', 1)
            ->where('school_year_id', AppSession::getSchoolYear())
            ->where('class_stage_id', $get('class_stage_id'))
            ->where('user_id', $get('ref_id'))
            ->pluck('name', 'id');
    })
    ->searchable()
    ->live()
    ->required()
    ->disabled(function ($get) {
        return is_null($get('class_stage_id'));
    }),
```

**Reply by leandrocfe** [reply, marked as answer]
I think ->hidden() or ->visible() would be more appropriate in this case because it will reload the field. Take a look

```php
->visible(function ($get): ?bool {
    return $get('class_stage_id');
})
```

afterStateUpdated should be

```php
->afterStateUpdated(function (Set $set, $state) {

    if ($state) {
        $set('class_room_id', null);
    }
})
```

**Reply by untillnesss** [reply]
Does that mean there's a problem with the filament?

**Comment by planetarquasar** [comment]
The problem still exists, any information about it? I know there was a marked answer, but I need this disabling logic, and not the hidden logic.

**Reply by leandrocfe** [reply]
Use `->searchable(fn ($get): ?bool => filled($get('class_stage_id')))`

**Reply by untillnesss** [reply]
I replaced it with `->visible()`

**Comment by maytham553-elite** [comment]
is there any solve with disabled please ?

**Reply by untillnesss** [reply]
Nope, maybe you should submit a new issue to get this noticed and resolved quickly.

**Reply by maytham553-elite** [reply]
i use the solution recommended there , https://github.com/filamentphp/filament/issues/3068
